### PR TITLE
Ignore files in deps dir when resolving wit packages

### PR DESCRIPTION
**Pull request by cardoso**
Some operating systems will add files randomly to directories.

Considering files inside `deps` currently serve no purpose and will cause an error on `UnresolvedPackage::parse_dir`, I think ignoring any files inside is the best option for user experience.

If that changes, it would be easy to tweak this filter to allow certain extensions.

### CODE REVIEW

File: crates/wit-parser/src/resolve.rs [hunk 1]
```diff
@@ -112,6 +112,12 @@ impl Resolve {
             for dep in path.read_dir().context("failed to read directory")? {
                 let dep = dep.context("failed to read directory iterator")?;
                 let path = dep.path();
+
+                // Files in deps dir are ignored
+                if path.is_file() {
+                    continue;
+                };
```
Comment (alexcrichton) [inline crates/wit-parser/src/resolve.rs:119]: This trailing semicolon can be dropped.

File: crates/wit-parser/src/resolve.rs [hunk 2]
```diff
@@ -112,6 +112,12 @@ impl Resolve {
             for dep in path.read_dir().context("failed to read directory")? {
                 let dep = dep.context("failed to read directory iterator")?;
                 let path = dep.path();
+
+                // Files in deps dir are ignored
```
Comment (alexcrichton) [inline crates/wit-parser/src/resolve.rs:116]: Could you expand this comment to explain why these files are being ignored?

File: .gitignore [hunk 3]
```diff
@@ -6,3 +6,6 @@ publish
 *.wasm
 *.wat
 test.config
+
+# Generated by macOS
+.DS_Store
```
Comment (alexcrichton) [inline .gitignore:11]: FWIW I think this is best stored in your global `~/.gitignore` file rather than having all projects you work on add this entry

**Review by alexcrichton — COMMENTED** [review]
Thanks! Could you add a test for this as well?

**Comment by cardoso** [comment]
Thanks for the review @alexcrichton . I made the changes and included a test.

**Review by alexcrichton — APPROVED** [review]

**Pull request merged**